lib/memo_wise.rb

```diff
@@ -423,6 +423,28 @@ def #{method_name}#{args_str}
           target.singleton_class.send(:private, method_name)
         end
       end
+
+      # Override [Module#instance_method](https://ruby-doc.org/core-3.0.0/Module.html#method-i-instance_method)
+      # to proxy the original `UnboundMethod#parameters` results. This proxying
+      # helps code that uses Ruby reflection over method parameters
+      def target.instance_method(symbol)
+        original_memo_wised_name = :"_memo_wise_original_#{symbol}"
```

Comment:
Should we pull this out since we use it elsewhere too? Wouldn't want the case where we change it somewhere and forget to change it here

Reply:
Can we delay that until a following PR when I extract all the private API stuff? I don't want to add additional helper methods now, as they pollute the method namespace of the target classes in the current implementation - I'm planning to change that soon -- and had already responded to you on different PRs that we could start extracting more, smaller helper methods at that point?

Reply:
Sounds good - sorry I'm losing track a bit of which PRs are still coming in this series and when we're extracting the helpers. As long as you have a good grasp of when we're doing this, works for me!

Reply:
Added a TODO for future extraction of the common method naming pattern